gas-tank/balance-services.md

```diff
@@ -0,0 +1,540 @@
+    message: "string", // [!code highlight]
+    signature: "string", // [!code highlight]
+  },
+})
+  .then((response) => {
+    // Balance consumed successfully
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+:::
+
+### Get a quote for consuming balance
+
+This endpoint allows users to generate a quote for consuming their balance on the Gas Tank platform.
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/balances/consume/quote`, {
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+  body: {
+    address: "string", // [!code highlight]
+    minDestinationAmount: "string", // [!code highlight]
+    destinationChain: "string", // [!code highlight]
+  },
+})
+  .then((response) => {
+    console.log(response);
+    // Handle & do something with the response
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+```json [Response]
+{
+  "expectedOutput": "string",
+  "fee": "string",
+  "spentAssets": [
+    {
+      "tokenAddress": "string",
+      "amount": "string",
+      "chain": "string"
+    }
+  ]
+}
+```
+
+:::
+
+### Get user balance queue update
+
+This endpoint allows users to get the balance queue update for a specific address.
+
+::: code-group
+
+```javascript [Request]
+const GAS_TANK_ENDPOINT = "https://gas-tank.xdefi.services";
+
+await fetch(`${GAS_TANK_ENDPOINT}/transactions`, {
+  method: "GET",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${jwtToken}`, // JWT token // [!code highlight]
+  },
+})
+  .then((response) => {
+    console.log(response);
+    // Handle & do something with the response
+  })
+  .catch((error) => {
+    // Catch & handle the error
+  });
+```
+
+```json [Response]
+{
+  "additionalProp1": [
+    {
+      "id": "string",
+      "type": "string",
+      "address": "string",
+      "tokenAddress": "string",
+      "amount": "string",
+      "chain": "string",
+      "destinationChain": "string",
+      "destinationAddress": "string",
+      "permitObject": {},
+      "processed": true,
+      "processedMetadata": {},
+      "pendingTransactions": [
+        {
+          "id": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
+          "chain": "ethereum",
+          "transactionHash": "0x6b175474e89094c44da98b954eedeac495271d0f",
+          "blockNumber": 123456,
+          "balanceUpdateId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
+          "confirmed": false,
+          "createdAt": "2023-01-01T00:00:00Z",
+          "updatedAt": "2023-01-01T00:00:00Z"
+        }
+      ],
+      "createdAt": "2024-04-11T04:13:07.102Z",
+      "updatedAt": "2024-04-11T04:13:07.102Z"
+    }
+  ],
+  ...
+}
+```
+
+:::
+
+### Get balance queue update by ID
+
+This endpoint allows users to get the balance queue update for a specific ID.
```

Comment:
can you explain that it is to get the status of any job ongoing (deposit, consume, internal transfer, withdrawal)